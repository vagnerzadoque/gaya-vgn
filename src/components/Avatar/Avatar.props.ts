import { IconName } from '@naturacosmeticos/natds-icons'
import { Size } from '@naturacosmeticos/natds-themes'
import { BrandTypes } from '../../brandTypes/brandTypes'

export type AvatarSizes = keyof Pick<Size, 'standard' | 'semi' | 'semiX' | 'medium' | 'largeXXX'>;

export type AvatarBaseProps = {
  /**
   * Specify an optional className to be added to your Avatar
   */
  className?: string
  /**
   * Specify an optional className to be added to your Avatar
   */
  brand?: BrandTypes

  /**
   * Optional ID for testing
   */
  testID?: string;

  /**
   * Specify the size of the Avatar
   * @default medium
   */
  size?: AvatarSizes
}

export type AvatarImageProps = AvatarBaseProps & {
  /**
   * Avatar type
   */
  type: 'image'

  /**
   * The alt text for the image
   */
  alt?: React.ImgHTMLAttributes<HTMLImageElement>['alt']

  /**
   * The source URL of the image you want to display
   */
  src?: React.ImgHTMLAttributes<HTMLImageElement>['src']

  /**
   * Event to fire if an error occurs while loading the image
   */
  onError?: React.ImgHTMLAttributes<HTMLImageElement>['onError']

}

export type AvatarIconProps = AvatarBaseProps & {
  /**
   * Avatar type
   */
  type: 'icon';

  /**
   * The icon name
   * @default outlined-default-mockup
   */
  name?: IconName;
}

export type AvatarLabelProps = AvatarBaseProps & {
  /**
   * Avatar type
   */
  type: 'label';

  /**
   * The label to be displayed in the Avatar.
   * It's highly recommended to pass a maximum of two characters
   * @default NA
   */
  label?: string;
}

export type AvatarProps = AvatarImageProps | AvatarIconProps | AvatarLabelProps;
