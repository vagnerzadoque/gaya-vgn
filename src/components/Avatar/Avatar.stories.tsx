/* eslint-disable no-param-reassign */
import React from 'react'
import type { Meta, StoryObj } from '@storybook/react'
import { Avatar, AvatarProps } from '.'

const componentStatus = `
> Avatar are used to identify a user

## Properties
| Property                      | Values                                    |    Status           |
|---                            |                                        ---|                  ---|
| **variant**                   | standard                                  | ✅ Available        |
| **type**                      | icon, label, image                        | ✅ Available        |
| **size**                      | standard, semi, semiX, medium, largeXXX   | ✅ Available        |
| **alt**                       | string                                    | ✅ Available        |
| **src**                       | string                                    | ✅ Available        |
| **onError**                   | function                                  | ✅ Available        |
| **name**                      | icon-name                                 | ✅ Available        |
| **brand**                   | avon, avon_v2, natura, natura_v2, theBodyShop, <br /> consultoriaDeBeleza, casaEestilo            | ✅ Available        |

## Technical Usages Examples
`

const handleError = (e: React.SyntheticEvent<HTMLImageElement, Event>) => {
  e.currentTarget.onerror = null
  e.currentTarget.src = 'https://via.placeholder.com/200x200'
}

const meta: Meta<AvatarProps> = {
  title: 'Components/Avatar',
  component: Avatar,
  parameters: {
    componentSubtitle: '',
    docs: { description: { component: componentStatus } }
  }
};

export default meta;
type Story = StoryObj<AvatarProps>;

export const Playground: Story = {
  args: {
    src: '/img_placeholder.png'
  },
  render: (args: AvatarProps) => <Avatar {...args} />
};

export const Types: Story = {
  render: () => (
    <div style={{ display: 'flex', gap: 8 }}>
      <Avatar type="image" src="/img_placeholder.png" onError={handleError} />
      <Avatar type="icon" />
      <Avatar type="label" />
    </div>
  )
};

export const Sizes: Story = {
  render: () => (
    <>
      <div style={{ display: 'flex', gap: 8, marginBottom: 8 }}>
        <Avatar type="image" src="/img_placeholder.png" size="standard" onError={handleError} />
        <Avatar type="image" src="/img_placeholder.png" size="semi" onError={handleError} />
        <Avatar type="image" src="/img_placeholder.png" size="semiX" onError={handleError} />
        <Avatar type="image" src="/img_placeholder.png" size="medium" onError={handleError} />
        <Avatar type="image" src="/img_placeholder.png" size="largeXXX" onError={handleError} />
      </div>
      <div style={{ display: 'flex', gap: 8, marginBottom: 8 }}>
        <Avatar type="icon" size="standard" />
        <Avatar type="icon" size="semi" />
        <Avatar type="icon" size="semiX" />
        <Avatar type="icon" size="medium" />
        <Avatar type="icon" size="largeXXX" />
      </div>
      <div style={{ display: 'flex', gap: 8, marginBottom: 8 }}>
        <Avatar type="label" size="standard" />
        <Avatar type="label" size="semi" />
        <Avatar type="label" size="semiX" />
        <Avatar type="label" size="medium" />
        <Avatar type="label" size="largeXXX" />
      </div>
    </>
  )
};
