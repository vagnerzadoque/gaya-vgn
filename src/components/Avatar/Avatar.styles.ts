import { createUseStyles } from 'react-jss'
import { Theme } from '@naturacosmeticos/natds-themes'
import { AvatarProps } from './Avatar.props'
import { buildTheme } from '../../ThemeProvider'

type AvatarStyleProps = Pick<AvatarProps, 'size' | 'brand'>

const getColorTheme = (theme: Theme) => ({ brand }: AvatarStyleProps) => {
  const themeSelectAvatar = buildTheme(brand, 'light')
  if (brand) {
    return {
      primary: themeSelectAvatar.color.primary,
      onPrimary: themeSelectAvatar.color.onPrimary
    }
  }
  return {
    primary: theme.color.primary,
    onPrimary: theme.color.onPrimary
  }
}

const styles = createUseStyles((theme: Theme) => ({
  surface: {
    display: 'flex',
    width: ({ size }: AvatarStyleProps) => (size && theme.size[size]),
    height: ({ size }: AvatarStyleProps) => (size && theme.size[size]),
    borderRadius: ({ size }: AvatarStyleProps) => (size && theme.avatar[size].borderRadius),
    backgroundColor: ({ brand }: AvatarStyleProps) => getColorTheme(theme)({ brand }).primary,
    justifyContent: 'center',
    alignItems: 'center',
    overflow: 'hidden'
  },
  image: {
    objectFit: 'cover',
    height: '100%'
  },
  label: {
    color: ({ brand }: AvatarStyleProps) => getColorTheme(theme)({ brand }).onPrimary,
    fontFamily: [theme.avatar.primary.fontFamily, theme.avatar.fallback.fontFamily],
    fontSize: ({ size }: AvatarStyleProps) => (size && theme.avatar[size].fontSize),
    fontWeight: [theme.avatar.primary.fontWeight, theme.avatar.fallback.fontWeight],
    letterSpacing: ({ size }: AvatarStyleProps) => (size && theme.avatar[size].letterSpacing),
    lineHeight: ({ size }: AvatarStyleProps) => (size && theme.avatar[size].lineHeight),
    padding: theme.spacing.micro,
    textTransform: 'uppercase'
  },
  icon: {
    padding: theme.spacing.micro
  }
}))

export default styles
