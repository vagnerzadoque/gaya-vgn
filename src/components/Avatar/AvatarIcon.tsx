import React from 'react'
import { IconName } from '@naturacosmeticos/natds-icons'
import { IconSize } from '../Icon/Icon.props'
import { AvatarIconProps, AvatarProps, AvatarSizes } from './Avatar.props'
import styles from './Avatar.styles'
import Icon from '../Icon'

export const isAvatarIcon = (props: AvatarProps): props is AvatarIconProps => props.type === 'icon'

export const convertIconSize = (size: AvatarSizes): string => {
  const iconSize = {
    standard: 'small',
    semi: 'standard',
    semiX: 'semi',
    medium: 'semiX',
    largeXXX: 'largeX'
  }

  return size && iconSize[size]
}

export const AvatarIcon = ({
  size = 'medium', className, brand, ...props
}: AvatarProps): JSX.Element => {
  const { icon } = styles({ size })

  return (
    <>
      {isAvatarIcon(props) && (
        <Icon
          brand={brand}
          className={`${className} ${icon}`}
          name={props.name as IconName}
          size={convertIconSize(size) as IconSize}
          color="onPrimary"
        />
      )}
    </>
  )
}

export default AvatarIcon
