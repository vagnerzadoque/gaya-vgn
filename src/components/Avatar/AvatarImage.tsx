import React from 'react'
import { AvatarImageProps, AvatarProps } from './Avatar.props'
import styles from './Avatar.styles'

export const isAvatarImage = (props: AvatarProps): props is AvatarImageProps => props.type === 'image'

export const AvatarImage = ({
  size = 'medium', className, ...props
}: AvatarProps) => {
  const { image } = styles({ size })

  return (
    <>
      {isAvatarImage(props) && (
        <img
          className={`${className} ${image}`}
          {...props}
        />
      )}
    </>
  )
}

export default AvatarImage
