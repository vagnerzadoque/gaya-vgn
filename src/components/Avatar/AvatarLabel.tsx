import React from 'react'
import { AvatarLabelProps, AvatarProps } from './Avatar.props'
import styles from './Avatar.styles'

export const isAvatarLabel = (props: AvatarProps): props is AvatarLabelProps => props.type === 'label'

export const AvatarLabel = ({ className = '', size = 'medium', ...props }: AvatarProps) => {
  const { label } = styles({ size })

  return (
    <>
      {isAvatarLabel(props) && (
        <label
          className={`${className} ${label}`}
          {...props}
        >
          {props.label ? props.label : 'NA'}
        </label>
      )}

    </>

  )
}

export default AvatarLabel
