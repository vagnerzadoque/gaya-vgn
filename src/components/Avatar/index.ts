import Avatar from './Avatar'

export { Avatar }
export { AvatarProps } from './Avatar.props'
export default Avatar
