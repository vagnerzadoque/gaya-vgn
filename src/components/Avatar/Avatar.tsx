import React from 'react'
import { AvatarProps } from './Avatar.props'
import styles from './Avatar.styles'
import AvatarImage from './AvatarImage'
import AvatarIcon from './AvatarIcon'
import AvatarLabel from './AvatarLabel'

export const renderVariant = (props: AvatarProps): React.ReactNode => {
  const { type } = props

  const types = {
    image: <AvatarImage {...props} />,
    label: <AvatarLabel {...props} />,
    icon: <AvatarIcon {...props} />
  }

  return types[type]
}

const Avatar = React.forwardRef<HTMLElement, AvatarProps>(
  (props, ref) => {
    const {
      className = '',
      size = 'medium',
      type = 'icon',
      brand,
      ...rest
    } = props
    const { surface } = styles({ size, brand })

    return (
      <span
        className={`${className} ${surface} Avatar-gaya`}
        ref={ref}
        {...rest}
      >
        {renderVariant({ ...props, size, type })}
      </span>
    )
  }
)

export default Avatar
