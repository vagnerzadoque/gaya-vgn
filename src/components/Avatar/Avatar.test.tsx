/* eslint-disable @typescript-eslint/ban-ts-comment */
import { fireEvent } from '@testing-library/react'
import * as React from 'react'
import Avatar from '.'
import renderWithTheme from '../../helpers/renderWithTheme'
import AvatarIcon, { convertIconSize } from './AvatarIcon'
import AvatarImage from './AvatarImage'
import AvatarLabel from './AvatarLabel'

describe('Avatar component', () => {
  it('should render correctly with default props', () => {
    // @ts-expect-error component without type
    const { styles, component } = renderWithTheme(<Avatar />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
  it('should render correctly with different size', () => {
    // @ts-expect-error component without type
    const { styles, component } = renderWithTheme(<Avatar size="semi" />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })

  describe('Label variant', () => {
    it('should render correctly when the variant is type label', () => {
      const component = renderWithTheme(<Avatar type="label" label="DS" />)
      const subcomponent = renderWithTheme(<AvatarLabel type="label" label="DS" />)

      expect([component.styles.toString(), component.component.container]).toMatchSnapshot()
      expect([subcomponent.styles.toString(), subcomponent.component.container]).toMatchSnapshot()
    })

    it('should render correctly when the variant is type label with default label', () => {
      const { styles, component } = renderWithTheme(<Avatar type="label" />)

      expect(component.container).toHaveTextContent('NA')
      expect([styles.toString(), component.container]).toMatchSnapshot()
    })
  })

  describe('Icon variant', () => {
    it('should render correctly when the variant is type icon', () => {
      const component = renderWithTheme(<Avatar type="icon" />)
      const subcomponent = renderWithTheme(<AvatarIcon type="icon" />)

      expect([component.toString(), component.component.container]).toMatchSnapshot()
      expect([subcomponent.toString(), subcomponent.component.container]).toMatchSnapshot()
    })
    it('should render correctly when the variant is type icon with the given iconName', () => {
      const { styles, component } = renderWithTheme(<Avatar type="icon" name="outlined-action-like" />)

      expect([styles.toString(), component.container]).toMatchSnapshot()
    })
  })

  describe('Image variant', () => {
    it('should render correctly with image variant', () => {
      const component = renderWithTheme(<Avatar type="image" src="image.jpg" alt="" />)
      const subcomponent = renderWithTheme(<AvatarImage type="image" src="image.jpg" alt="" />)

      expect([component.styles.toString(), component.component.container]).toMatchSnapshot()
      expect([subcomponent.styles.toString(), subcomponent.component.container]).toMatchSnapshot()
    })
    it('should fire onError when the image src is broken', () => {
      const onErrorMock = jest.fn()
      const { styles, component } = renderWithTheme(<Avatar type="image" src="image.junk" alt="image" onError={onErrorMock} />)

      fireEvent.error(component.getByAltText('image'))

      expect([styles.toString(), component.container]).toMatchSnapshot()
      expect(onErrorMock).toHaveBeenCalled()
    })
  })
})

describe('convertIconSize', () => {
  it('should return the small size when the input is standard', () => {
    expect(convertIconSize('standard')).toBe('small')
  })
})
