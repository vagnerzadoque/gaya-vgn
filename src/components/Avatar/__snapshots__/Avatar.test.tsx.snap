// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component Icon variant should render correctly when the variant is type icon 1`] = `
[
  "[object Object]",
  <div>
    <span
      class=" surface-0-2-1 surface-d0-0-2-5 Avatar-gaya"
    >
      <i
        aria-hidden="true"
        class="undefined icon-0-2-4 icon-0-2-9 icon-d0-0-2-10 natds-icons natds-icons-outlined-default-mockup Icon-gaya"
        data-testid="icon-undefined"
        role="img"
      />
    </span>
  </div>,
]
`;

exports[`Avatar component Icon variant should render correctly when the variant is type icon 2`] = `
[
  "[object Object]",
  <div>
    <i
      aria-hidden="true"
      class="undefined icon-0-2-4 icon-0-2-7 icon-d0-0-2-8 natds-icons natds-icons-outlined-default-mockup Icon-gaya"
      data-testid="icon-undefined"
      role="img"
    />
  </div>,
]
`;

exports[`Avatar component Icon variant should render correctly when the variant is type icon with the given iconName 1`] = `
[
  ".surface-0-2-1 {
  display: flex;
  overflow: hidden;
  align-items: center;
  justify-content: center;
}
.image-0-2-2 {
  height: 100%;
  object-fit: cover;
}
.label-0-2-3 {
  padding: 4px;
  font-family: Roboto, Arial;
  font-weight: 400, 400;
  text-transform: uppercase;
}
.icon-0-2-4 {
  padding: 4px;
}
.surface-d0-0-2-5 {
  width: 48px;
  height: 48px;
  border-radius: 24px;
  background-color: #f4ab34;
}
.label-d1-0-2-6 {
  color: #333333;
  font-size: 24px;
  letter-spacing: 0;
  line-height: 1.5;
}
.surface-d2-0-2-7 {
  width: 48px;
  height: 48px;
  border-radius: 24px;
  background-color: #f4ab34;
}
.label-d3-0-2-8 {
  color: #333333;
  font-size: 24px;
  letter-spacing: 0;
  line-height: 1.5;
}
.icon-0-2-9 {
  font-family: natds-icons;
  user-select: none;
  pointer-events: none;
}
.icon-d0-0-2-10 {
  color: #333333;
  font-size: 40px;
}",
  <div>
    <span
      class=" surface-0-2-1 surface-d0-0-2-5 Avatar-gaya"
      name="outlined-action-like"
    >
      <i
        aria-hidden="true"
        class="undefined icon-0-2-4 icon-0-2-9 icon-d0-0-2-10 natds-icons natds-icons-outlined-action-like Icon-gaya"
        data-testid="icon-outlined-action-like"
        role="img"
      />
    </span>
  </div>,
]
`;

exports[`Avatar component Image variant should fire onError when the image src is broken 1`] = `
[
  ".surface-0-2-1 {
  display: flex;
  overflow: hidden;
  align-items: center;
  justify-content: center;
}
.image-0-2-2 {
  height: 100%;
  object-fit: cover;
}
.label-0-2-3 {
  padding: 4px;
  font-family: Roboto, Arial;
  font-weight: 400, 400;
  text-transform: uppercase;
}
.icon-0-2-4 {
  padding: 4px;
}
.surface-d0-0-2-5 {
  width: 48px;
  height: 48px;
  border-radius: 24px;
  background-color: #f4ab34;
}
.label-d1-0-2-6 {
  color: #333333;
  font-size: 24px;
  letter-spacing: 0;
  line-height: 1.5;
}
.surface-d2-0-2-7 {
  width: 48px;
  height: 48px;
  border-radius: 24px;
  background-color: #f4ab34;
}
.label-d3-0-2-8 {
  color: #333333;
  font-size: 24px;
  letter-spacing: 0;
  line-height: 1.5;
}",
  <div>
    <span
      alt="image"
      class=" surface-0-2-1 surface-d0-0-2-5 Avatar-gaya"
      src="image.junk"
    >
      <img
        alt="image"
        class="undefined image-0-2-2"
        src="image.junk"
        type="image"
      />
    </span>
  </div>,
]
`;

exports[`Avatar component Image variant should render correctly with image variant 1`] = `
[
  ".surface-0-2-1 {
  display: flex;
  overflow: hidden;
  align-items: center;
  justify-content: center;
}
.image-0-2-2 {
  height: 100%;
  object-fit: cover;
}
.label-0-2-3 {
  padding: 4px;
  font-family: Roboto, Arial;
  font-weight: 400, 400;
  text-transform: uppercase;
}
.icon-0-2-4 {
  padding: 4px;
}
.surface-d0-0-2-5 {
  width: 48px;
  height: 48px;
  border-radius: 24px;
  background-color: #f4ab34;
}
.label-d1-0-2-6 {
  color: #333333;
  font-size: 24px;
  letter-spacing: 0;
  line-height: 1.5;
}
.surface-d2-0-2-7 {
  width: 48px;
  height: 48px;
  border-radius: 24px;
  background-color: #f4ab34;
}
.label-d3-0-2-8 {
  color: #333333;
  font-size: 24px;
  letter-spacing: 0;
  line-height: 1.5;
}",
  <div>
    <span
      alt=""
      class=" surface-0-2-1 surface-d0-0-2-5 Avatar-gaya"
      src="image.jpg"
    >
      <img
        alt=""
        class="undefined image-0-2-2"
        src="image.jpg"
        type="image"
      />
    </span>
  </div>,
]
`;

exports[`Avatar component Image variant should render correctly with image variant 2`] = `
[
  ".surface-0-2-1 {
  display: flex;
  overflow: hidden;
  align-items: center;
  justify-content: center;
}
.image-0-2-2 {
  height: 100%;
  object-fit: cover;
}
.label-0-2-3 {
  padding: 4px;
  font-family: Roboto, Arial;
  font-weight: 400, 400;
  text-transform: uppercase;
}
.icon-0-2-4 {
  padding: 4px;
}
.surface-d0-0-2-5 {
  width: 48px;
  height: 48px;
  border-radius: 24px;
  background-color: #f4ab34;
}
.label-d1-0-2-6 {
  color: #333333;
  font-size: 24px;
  letter-spacing: 0;
  line-height: 1.5;
}",
  <div>
    <img
      alt=""
      class="undefined image-0-2-2"
      src="image.jpg"
      type="image"
    />
  </div>,
]
`;

exports[`Avatar component Label variant should render correctly when the variant is type label 1`] = `
[
  ".surface-0-2-1 {
  display: flex;
  overflow: hidden;
  align-items: center;
  justify-content: center;
}
.image-0-2-2 {
  height: 100%;
  object-fit: cover;
}
.label-0-2-3 {
  padding: 4px;
  font-family: Roboto, Arial;
  font-weight: 400, 400;
  text-transform: uppercase;
}
.icon-0-2-4 {
  padding: 4px;
}
.surface-d0-0-2-5 {
  width: 48px;
  height: 48px;
  border-radius: 24px;
  background-color: #f4ab34;
}
.label-d1-0-2-6 {
  color: #333333;
  font-size: 24px;
  letter-spacing: 0;
  line-height: 1.5;
}
.surface-d2-0-2-7 {
  width: 48px;
  height: 48px;
  border-radius: 24px;
  background-color: #f4ab34;
}
.label-d3-0-2-8 {
  color: #333333;
  font-size: 24px;
  letter-spacing: 0;
  line-height: 1.5;
}",
  <div>
    <span
      class=" surface-0-2-1 surface-d0-0-2-5 Avatar-gaya"
      label="DS"
    >
      <label
        class=" label-0-2-3 label-d3-0-2-8"
        label="DS"
        type="label"
      >
        DS
      </label>
    </span>
  </div>,
]
`;

exports[`Avatar component Label variant should render correctly when the variant is type label 2`] = `
[
  ".surface-0-2-1 {
  display: flex;
  overflow: hidden;
  align-items: center;
  justify-content: center;
}
.image-0-2-2 {
  height: 100%;
  object-fit: cover;
}
.label-0-2-3 {
  padding: 4px;
  font-family: Roboto, Arial;
  font-weight: 400, 400;
  text-transform: uppercase;
}
.icon-0-2-4 {
  padding: 4px;
}
.surface-d0-0-2-5 {
  width: 48px;
  height: 48px;
  border-radius: 24px;
  background-color: #f4ab34;
}
.label-d1-0-2-6 {
  color: #333333;
  font-size: 24px;
  letter-spacing: 0;
  line-height: 1.5;
}",
  <div>
    <label
      class=" label-0-2-3 label-d1-0-2-6"
      label="DS"
      type="label"
    >
      DS
    </label>
  </div>,
]
`;

exports[`Avatar component Label variant should render correctly when the variant is type label with default label 1`] = `
[
  ".surface-0-2-1 {
  display: flex;
  overflow: hidden;
  align-items: center;
  justify-content: center;
}
.image-0-2-2 {
  height: 100%;
  object-fit: cover;
}
.label-0-2-3 {
  padding: 4px;
  font-family: Roboto, Arial;
  font-weight: 400, 400;
  text-transform: uppercase;
}
.icon-0-2-4 {
  padding: 4px;
}
.surface-d0-0-2-5 {
  width: 48px;
  height: 48px;
  border-radius: 24px;
  background-color: #f4ab34;
}
.label-d1-0-2-6 {
  color: #333333;
  font-size: 24px;
  letter-spacing: 0;
  line-height: 1.5;
}
.surface-d2-0-2-7 {
  width: 48px;
  height: 48px;
  border-radius: 24px;
  background-color: #f4ab34;
}
.label-d3-0-2-8 {
  color: #333333;
  font-size: 24px;
  letter-spacing: 0;
  line-height: 1.5;
}",
  <div>
    <span
      class=" surface-0-2-1 surface-d0-0-2-5 Avatar-gaya"
    >
      <label
        class=" label-0-2-3 label-d3-0-2-8"
        type="label"
      >
        NA
      </label>
    </span>
  </div>,
]
`;

exports[`Avatar component should render correctly with default props 1`] = `
[
  ".surface-0-2-1 {
  display: flex;
  overflow: hidden;
  align-items: center;
  justify-content: center;
}
.image-0-2-2 {
  height: 100%;
  object-fit: cover;
}
.label-0-2-3 {
  padding: 4px;
  font-family: Roboto, Arial;
  font-weight: 400, 400;
  text-transform: uppercase;
}
.icon-0-2-4 {
  padding: 4px;
}
.surface-d0-0-2-5 {
  width: 48px;
  height: 48px;
  border-radius: 24px;
  background-color: #f4ab34;
}
.label-d1-0-2-6 {
  color: #333333;
  font-size: 24px;
  letter-spacing: 0;
  line-height: 1.5;
}
.surface-d2-0-2-7 {
  width: 48px;
  height: 48px;
  border-radius: 24px;
  background-color: #f4ab34;
}
.label-d3-0-2-8 {
  color: #333333;
  font-size: 24px;
  letter-spacing: 0;
  line-height: 1.5;
}
.icon-0-2-9 {
  font-family: natds-icons;
  user-select: none;
  pointer-events: none;
}
.icon-d0-0-2-10 {
  color: #333333;
  font-size: 40px;
}",
  <div>
    <span
      class=" surface-0-2-1 surface-d0-0-2-5 Avatar-gaya"
    >
      <i
        aria-hidden="true"
        class="undefined icon-0-2-4 icon-0-2-9 icon-d0-0-2-10 natds-icons natds-icons-outlined-default-mockup Icon-gaya"
        data-testid="icon-undefined"
        role="img"
      />
    </span>
  </div>,
]
`;

exports[`Avatar component should render correctly with different size 1`] = `
[
  ".surface-0-2-1 {
  display: flex;
  overflow: hidden;
  align-items: center;
  justify-content: center;
}
.image-0-2-2 {
  height: 100%;
  object-fit: cover;
}
.label-0-2-3 {
  padding: 4px;
  font-family: Roboto, Arial;
  font-weight: 400, 400;
  text-transform: uppercase;
}
.icon-0-2-4 {
  padding: 4px;
}
.surface-d0-0-2-5 {
  width: 32px;
  height: 32px;
  border-radius: 16px;
  background-color: #f4ab34;
}
.label-d1-0-2-6 {
  color: #333333;
  font-size: 14px;
  letter-spacing: 0.22px;
  line-height: 1.5;
}
.surface-d2-0-2-7 {
  width: 32px;
  height: 32px;
  border-radius: 16px;
  background-color: #f4ab34;
}
.label-d3-0-2-8 {
  color: #333333;
  font-size: 14px;
  letter-spacing: 0.22px;
  line-height: 1.5;
}
.icon-0-2-9 {
  font-family: natds-icons;
  user-select: none;
  pointer-events: none;
}
.icon-d0-0-2-10 {
  color: #333333;
  font-size: 24px;
}",
  <div>
    <span
      class=" surface-0-2-1 surface-d0-0-2-5 Avatar-gaya"
    >
      <i
        aria-hidden="true"
        class="undefined icon-0-2-4 icon-0-2-9 icon-d0-0-2-10 natds-icons natds-icons-outlined-default-mockup Icon-gaya"
        data-testid="icon-undefined"
        role="img"
      />
    </span>
  </div>,
]
`;
